src/TextInputMask.tsx

```diff
@@ -7,6 +7,10 @@ function detectCharacter(mask: string): string {
   return c || ''
 }
 
+function escapeForRegExp(value: string): string {
+  return value.replace(/[.*+?^${}()|[\]\\]/g, '\\$&')
```

Comment:
`dot` is the only special character that is used here for `splitCharacter`. I added all special characters here because maybe this can be abstracted to util in the future and used in other places. If you prefer to have only `.`, I can change this.

Reply:
From https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_expressions#escaping